client/src/components/HomePage/LandingPage/Filter/index.js

```diff
@@ -1,9 +1,110 @@
+/* eslint-disable react/prop-types */
 import React, { Component } from 'react';
 import './index.css';
 
 class Filter extends Component {
+  state = {
+    categoryList: [
+      'General',
+      'Education',
+      'Health',
+      'Disability',
+      'Overseas Aid',
+      'Poverty Relief',
+      'Housing',
+      'Religious activities',
+      'Art/culture',
+      'Amateur sport',
+      'Animals',
+      'Environment',
+      'Economic',
+      'Armed Forces',
+      'Human rights',
+      'Recreation',
+      'Others',
+    ],
+    fromInc: -1,
+    toInc: -1,
+    fromEx: -1,
+    toEx: -1,
+    income: false,
+    expend: false,
+    category: false,
+    valueSelect: -1,
+    error: '',
+  };
+
+  handleChange = event => {
+    const { target } = event;
+    let values;
+    if (target.type === 'checkbox') {
+      values = target.checked;
+    } else {
+      values = target.value;
+    }
+    this.setState({ [target.name]: values });
+  };
+
+  handleSelect = event => {
+    this.setState({ valueSelect: event.target.value });
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+    let url = '';
+    const {
+      fromInc,
+      toInc,
+      fromEx,
+      toEx,
+      income,
+      expend,
+      category,
+      valueSelect,
+      error,
+    } = this.state;
+
+    if (income || expend) {
+      if (fromInc > toInc || fromEx > toEx) {
+        return this.setState({ error: 'Error Input To less than Form ' });
+      }
+    }
+    if (income && expend && category) {
+      url = `/search?incfrom=${fromInc}&incto=${toInc}&exform=${fromEx}&exto=${toEx}&category='${valueSelect}'`;
+    } else if (income && category) {
+      url = `/search?incfrom=${fromInc}&incto=${toInc}&exform=-1&exto=-1&category='${valueSelect}'`;
+    } else if (expend && category) {
+      url = `/search?incfrom=-1&incto=-1&exform=${fromEx}&exto=${toEx}&category='${valueSelect}'`;
+    } else if (income && expend) {
+      url = `/search?incfrom=${fromInc}&incto=${toInc}&exform=${fromEx}&exto=${toEx}&category=-1`;
+    } else if (income) {
+      url = `/search?incfrom=${fromInc}&incto=${toInc}&exform=-1&exto=-1&category=-1`;
+    } else if (expend) {
+      url = `/search?incfrom=-1&incto=-1&exform=${fromEx}&exto=${toEx}&category=-1`;
+    } else if (category) {
+      url = `/search?incfrom=-1&incto=-1&exform=-1&exto=-1&category='${valueSelect}'`;
+    } else {
+      url = `/search?incfrom=-1&incto=-1&exform=-1&exto=-1&category=-1`;
+    }
+    const { history } = this.props;
```

Comment:
@Asmaa-ezz 
react-router has a HOC called [withRouter](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/withRouter.md) that lets you use this.props.history in a component without having to pass it all the way down to that component.  It might be helpful here.  You'd import withRouter from react-router in this Filter component and then at the bottom of the file you'd export it wrapped in withRouter(), like this:
`export default withRouter(Filter);`
Then it gives you this.props.history (and this.props.location and this.props.match too) for "free", without you having to pass in the history prop in all of the in-between components (like LandingPage and Homepage).